servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -778,6 +779,8 @@ public void schedule(final Throwable originalCause) {
                                 host.healthCheckConfig.executor)
                                 .apply(0, originalCause)
                                 .concat(connectionFactory.newConnection(host.address, null, null)
+                                        // Remove state from async context
+                                        .afterOnSubscribe(__ -> AsyncContext.clear())
```

Comment:
`afterOnSubscribe` is too late in case users access `AsyncContext` from `beforeSubscriber` or `beforeOnSubscribe`. 

Consider either `beforeOnSubscribe` placed before `concat` or wrap the whole thing with `defer` and clear the context there.